tests/components/amberelectric/test_coordinator.py

```diff
@@ -200,3 +209,36 @@ async def test_fetch_general_and_feed_in_site(
         FEED_IN_CHANNEL[3],
     ]
     assert result["grid"]["renewables"] == round(GENERAL_CHANNEL[0].renewables)
+    assert result["grid"]["price_spike"] == "none"
+
+
+async def test_fetch_potential_spike(
+    hass: HomeAssistant, current_price_api: Mock
+) -> None:
+    """Test fetching a site with only a general channel."""
+
+    general_channel: list[CurrentInterval] = [
+        generate_current_interval(
+            ChannelType.GENERAL, parser.parse("2021-09-21T08:30:00+10:00")
+        ),
+    ]
+    general_channel[0].spike_status = SpikeStatus.POTENTIAL
+    current_price_api.get_current_price.return_value = general_channel
+    data_service = AmberUpdateCoordinator(hass, current_price_api, GENERAL_ONLY_SITE_ID)
```

Comment:
Please don't access integration details like the coordinator in the tests. We only want to assert core state.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations